Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Multistage docker build, requires docker 17.05
+
+# builder stage
+FROM nvidia/cuda:10.0-devel as builder
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get --no-install-recommends --yes install \
+        libncurses5-dev \
+        libncursesw5-dev \
+        cmake \
+        git \
+        curl
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+
+RUN git clone https://github.com/mimblewimble/grin-miner && cd grin-miner && git submodule update --init
+
+RUN cd grin-miner && sed -i '/cuckoo-miner" }/s/^/#/' Cargo.toml && sed -i '/^#.*build-cuda-plugins"]/s/^#//' Cargo.toml
+
+RUN cd grin-miner && $HOME/.cargo/bin/cargo build --release
+
+# runtime stage
+FROM nvidia/cuda:10.0-base
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get --no-install-recommends --yes install \
+    libncurses5 \
+    libncursesw5
+
+COPY --from=builder /grin-miner/target/release/grin-miner /grin-miner/target/release/grin-miner
+COPY --from=builder /grin-miner/target/release/plugins/* /grin-miner/target/release/plugins/
+COPY --from=builder /grin-miner/grin-miner.toml /grin-miner/grin-miner.toml
+
+WORKDIR /grin-miner
+
+RUN sed -i -e 's/run_tui = true/run_tui = false/' grin-miner.toml
+
+RUN echo $'#!/bin/bash\n\
```

Comment:
I checked the generated run.sh file and it starts with `$#!/bin/bash`
I suggest to remove `$`:

    tredaell@aldebaran grin-miner$ docker run -ti --rm grin-miner
    run.sh: 1: run.sh: 0!/bin/bash: not found

Tested on `Docker version 18.09.0, build 4d60db4`

Reply:
was a typo. Removed